### DISCUSSION TITLE
Help Needed for Alarm Manager

### DISCUSSION THREAD

**Discussion by iraunit** [WrichikBasu/ShakeAlarmClock#52] [General]
I am learning Android and trying to automatically wake up the phone if it is in sleep mode and show something like an alarm app shows from a broadcast Receiver. I am not finding this anywhere, I think you have implemented this feature, I also tried searching in the code but couldn't find it. Could you please help me?

**Comment by WrichikBasu** [comment]
Sorry for being so late; somehow I missed the email from GitHub. The code to set the alarm can be found in the following method:

https://github.com/WrichikBasu/ShakeAlarmClock/blob/3baae64fe5a8c5887ef41c8632394fe5ec5891df/app/src/main/java/in/basulabs/shakealarmclock/Activity_AlarmsList.java#L505

The intent is delivered to the broadcast receiver, which then acts accordingly.

Note that you need the `android.permission.SCHEDULE_EXACT_ALARM` in Manifest (assuming you are targetting Android 13) to be able to schedule exact alarms. Furthermore, Android is very specific about setting exact alarms, and they can only be set by apps which require them for their core functionality.

This article may be helpful, though it is written for Android 12:
https://proandroiddev.com/quick-guide-of-alarms-on-android-12-3a864ea81910
